src/airflow_tools/providers/filesystem/operators/filesystem.py

```diff
@@ -53,3 +54,58 @@ def execute(self, context):
             else self.destination_path
         )
         destination_fs_hook.write(data, full_destination_path)
+
+
+class SQLToFilesystem(BaseOperator):
+    """
+    Copies data from a SQL query to a filesystem.
+    """
+
+    template_fields = (
+        'source_name',
+        'source_schema_name',
+        'source_table_name',
+        'destination_path',
+    )
+
+    def __init__(
+        self,
+        source_sql_conn_id: str,
+        destination_fs_conn_id: str,
+        source_name: str,
+        source_schema_name: str,
+        source_table_name: str,
+        destination_path: str,
+        batch_size: Optional[int] = 100000,
+        *args,
+        **kwargs,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.source_sql_conn_id = source_sql_conn_id
+        self.destination_fs_conn_id = destination_fs_conn_id
+        self.source_name = source_name
+        self.source_schema_name = source_schema_name
+        self.source_table_name = source_table_name
+        self.destination_path = destination_path
+        self.batch_size = batch_size
+
+    def execute(self, context):
+        source_sql_hook: DbApiHook = BaseHook.get_connection(
+            self.source_sql_conn_id
+        ).get_hook()
+        source_sql_hook.__schema__ = self.source_schema_name
+
+        destination_fs_hook = FilesystemFactory.get_data_lake_filesystem(
+            connection=BaseHook.get_connection(self.destination_fs_conn_id),
+        )
+
+        query = f'SELECT * FROM {self.source_table_name};'
+        for i, df in enumerate(
+            source_sql_hook.get_pandas_df_by_chunks(
+                sql=query, chunksize=self.batch_size
+            ),
+            start=1,
+        ):
+            generated_path = f"{self.source_name.lower()}/{self.source_table_name.lower()}/part{i:04}.parquet"
```

Comment:
I would prefer if we are not opinionated on the path and let the user choose the full path. If they want to have the source table name they should define it in the template. I think we should only be adding the parquet file name at the end of the `self.destination_path` that the user provided.